uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Helper/ContactActionsMenu/LeaveActionGroup.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+use CRM_HRContactActionsMenu_Component_Group as ActionsGroup;
+use CRM_HRContactActionsMenu_Component_GroupButtonItem as ActionsGroupButtonItem;
+use CRM_HRContactActionsMenu_Component_GroupSeparatorItem as GroupSeparatorItem;
+use CRM_HRLeaveAndAbsences_Service_LeaveManager as LeaveManagerService;
+use CRM_HRLeaveAndAbsences_Component_ContactActionsMenu_NoSelectedLeaveApproverItem as NoSelectedLeaveApproverItem;
+use CRM_HRLeaveAndAbsences_Component_ContactActionsMenu_LeaveApproversListItem as LeaveApproversListItem;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Helper_ContactActionsMenu_LeaveActionGroup
+ */
+class CRM_HRLeaveAndAbsences_Helper_ContactActionsMenu_LeaveActionGroup {
+
+  use CRM_HRLeaveAndAbsences_Service_SettingsManagerTrait;
+
+  /**
+   * @var LeaveManagerService
+   */
+  private $leaveManagerService;
+
+  /**
+   * @var int
+   */
+  private $contactID;
+
+  /**
+   * CRM_HRLeaveAndAbsences_Helper_ContactActionsMenu_LeaveActionGroup constructor.
+   *
+   * @param \CRM_HRLeaveAndAbsences_Service_LeaveManager $leaveManagerService
+   * @param int $contactID
+   */
+  public function __construct(LeaveManagerService $leaveManagerService, $contactID) {
+    $this->leaveManagerService = $leaveManagerService;
+    $this->contactID = $contactID;
+  }
+
+  /**
+   * Gets Leave Menu Group with menu items already
+   * added.
+   *
+   * @return ActionsGroup
+   */
+  public function get() {
+    $actionsGroup = new ActionsGroup('Leave:');
+    $actionsGroup->addItem($this->getRecordLeaveButton());
+    $actionsGroup->addItem($this->getRecordSicknessButton());
+    $actionsGroup->addItem($this->getRecordOvertimeButton());
+    $actionsGroup->addItem($this->getViewEntitlementsButton());
+    $actionsGroup->addItem(new GroupSeparatorItem());
+
+    $leaveApprovers = $this->getLeaveApprovers();
+
+    if ($leaveApprovers) {
+      $leaveApproversListItem = new LeaveApproversListItem($this->leaveManagerService, $this->contactID);
```

Comment:
Wouldn't it be better to pass the list of approvers to `LeaveApproversListItem`? This way you would avoid getting the list twice

Reply:
Good Idea